Portal displays "Internal 500 error" (mysql) when doing group membership operations

**Describe the bug**
When adding or deleting users, or toggling Group Manager for users, the portal displays an "HTTP 500 (Internal Server Error): There was an internal server error." banner.

**VinylDNS Version**
0.9.4-SNAPSHOT

**To Reproduce**
Steps to reproduce the behavior:
1. Click on Groups
1. Click View for an existing group (ex. `testGroup`)
2. Enter username (uid) in 'NT User Name' field and click 'Add Group Member'
3. User is added to group; HTTP 500 error banner is displayed at top of portal page
4. Click `Group Manager` toggle; HTTP 500 error banner is displayed at top of portal page
5. Click 'Delete' button for user; HTTP 500 error banner is displayed at top of portal page

Latent Effects:
1. Even when deleted from group, group shows in user's group list, with no interaction buttons ('View' 'Edit' 'Delete')
2. Once a user is added and deleted from an admin group for a zone, user has full read,write,edit access for zone, even if ACL restricting user is added later.

**Expected behavior**
Expected behavior is "user" is added to and removed from groups with no issues; with group membership displays and ACLs being respected.

**Screenshots**
none

**Stack trace or error log output**
`docker logs vinyldns-api`:
```
2019-10-08 15:50:33,336 [test] ERROR | (a.a.ActorSystemImpl:73) | Error during processing of request: '(conn=23814) You have an error in your SQL syntax; check the manual that corresponds to your MySQ
L server version for the right syntax to use near ')' at line 1'. Completing with 500 Internal Server Error response. To change default exception handling behavior, provide a custom ExceptionHandler. 
java.sql.SQLSyntaxErrorException: (conn=23814) You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')' at line 1
--- 8< --- snip ---
Caused by: java.sql.SQLException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1
Query is: INSERT IGNORE INTO membership (user_id, group_id) VALUES 
java thread: VinylDNS-akka.actor.default-dispatcher-335
        at org.mariadb.jdbc.internal.util.LogQueryTool.exceptionWithQuery(LogQueryTool.java:163) ~[org.mariadb.jdbc.mariadb-java-client-2.3.0.jar:na]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:280) ~[org.mariadb.jdbc.mariadb-java-client-2.3.0.jar:na]
        at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeInternal(MariaDbPreparedStatementClient.java:223) ~[org.mariadb.jdbc.mariadb-java-client-2.3.0.jar:na]
        ... 79 common frames omitted
```


**Additional context**
Current test configuration is:
```
CONTAINER ID        IMAGE                            COMMAND                  CREATED             STATUS              PORTS                                NAMES
51007ee207b3        vinyldns/api:0.9.4-SNAPSHOT      "/opt/docker/bin/api"    25 minutes ago      Up 25 minutes       0.0.0.0:9000->9000/tcp               vinyldns-api
0fdc91ec1abb        mysql:5.7                        "docker-entrypoint..."   4 days ago          Up 4 days           33060/tcp, 0.0.0.0:19002->3306/tcp   vinyldns-mysql
1da684bf8d12        vinyldns/portal:0.9.4-SNAPSHOT   "/opt/docker/bin/p..."   4 days ago          Up About an hour    0.0.0.0:9001->9001/tcp               vinyldns-portal
```

api/docker.conf:
```
vinyldns {

  data-stores = ["mysql"]

  mysql {
    settings {
      name = "vinyldns"
      driver = "org.mariadb.jdbc.Driver"
      migration-url = "jdbc:mariadb://vinyldns-mysql:3306/?user=root&password=pass"
      url = "jdbc:mariadb://vinyldns-mysql:3306/vinyldns?user=root&password=pass"
      user = "root"
      password = "pass"

      connection-timeout-millis=5000

      max-lifetime = 900000
      maximum-pool-size = 30
      register-mbeans = true

      idle-timeout = 10000
      minimum-idle = 20

      # these database settings are recommended here https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration
      # as well as here http://assets.en.oreilly.com/1/event/21/Connector_J%20Performance%20Gems%20Presentation.pdf
      # and here https://www.facebook.com/notes/mysql-at-facebook/jdbc-connection-overheads/10152027251270933/
      my-sql-properties = {
        cacheServerConfiguration=true
        prepStmtCacheSize=50
        prepStmtCacheSqlLimit=2048
        cachePrepStmts=true
        useServerPrepStmts=true
        useLocalSessionState=true
        useLocalTransactionState=true
        cacheResultSetMetadata=true
        rewriteBatchedStatements=true
      }

    }

    repositories {
      batch-change {
      }
      record-set {
      }
      user {
      }
      zone {
      }
      record-change {
      }
      zone-change {
      }
      group {
      }
      group-change {
      }
      membership {
      }
    }
  }

  queue {
    class-name = "vinyldns.mysql.queue.MySqlMessageQueueProvider"

    polling-interval = 250.millis
    messages-per-poll = 10

    settings = {
      name = "vinyldns"
      driver = "org.mariadb.jdbc.Driver"
      migration-url = "jdbc:mariadb://vinyldns-mysql:3306/?user=root&password=pass"
      url = "jdbc:mariadb://vinyldns-mysql:3306/vinyldns?user=root&password=pass"
      user = "root"
      password = "pass"

      # see https://github.com/brettwooldridge/HikariCP
      connection-timeout-millis = 1000
      idle-timeout = 10000
      max-lifetime = 30000
      maximum-pool-size = 5
      minimum-idle = 1

      my-sql-properties = {
        cachePrepStmts=true
        prepStmtCacheSize=250
        prepStmtCacheSqlLimit=2048
        rewriteBatchedStatements=true
      }
    }
  }

...

```

--- a/modules/dynamodb/src/it/scala/vinyldns/dynamodb/repository/DynamoDBMembershipRepositoryIntegrationSpec.scala
+++ b/modules/dynamodb/src/it/scala/vinyldns/dynamodb/repository/DynamoDBMembershipRepositoryIntegrationSpec.scala
@@ -54,7 +54,18 @@ class DynamoDBMembershipRepositoryIntegrationSpec extends DynamoDBIntegrationSpe
     val user1 = genString
     val user2 = genString
     "add members successfully" in {
-      repo.addMembers(groupId, Set(user1, user2)).unsafeRunSync() should contain theSameElementsAs Set(user1, user2)
+      repo
+        .addMembers(groupId, Set(user1, user2))
+        .unsafeRunSync() should contain theSameElementsAs Set(user1, user2)
+    }
+
+    "add members with no member ids invokes no change" in {
+      val user1 = genString
+      repo.addMembers(groupId, Set(user1)).unsafeRunSync()
+
+      val originalResult = repo.getGroupsForUser(user1).unsafeRunSync()
+      repo.addMembers(groupId, Set()).unsafeRunSync()
+      repo.getGroupsForUser(user1).unsafeRunSync() should contain theSameElementsAs originalResult
     }
 
     "add a group to an existing user" in {
@@ -102,6 +113,15 @@ class DynamoDBMembershipRepositoryIntegrationSpec extends DynamoDBIntegrationSpe
         } yield userGroups
 
       f.unsafeRunSync() shouldBe empty
+    }
+
+    "remove members with no member ids invokes no change" in {
+      val user1 = genString
+      repo.addMembers(groupId, Set(user1)).unsafeRunSync()
+
+      val originalResult = repo.getGroupsForUser(user1).unsafeRunSync()
+      repo.removeMembers(groupId, Set()).unsafeRunSync()
+      repo.getGroupsForUser(user1).unsafeRunSync() should contain theSameElementsAs originalResult
     }
 
     "remove all groups for user" in {

--- a/modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala
+++ b/modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala
@@ -57,6 +57,17 @@ class MySqlMembershipRepositoryIntegrationSpec
     }
 
   "MySqlMembershipRepo.addMembers" should {
+    "do nothing if member ids is empty" in {
+      val originalResult = getAllRecords
+      val groupId = "group-id-1"
+      val userIds: Set[String] = Set()
+      val addResult = repo.addMembers(groupId, userIds).unsafeRunSync()
+      addResult shouldBe empty
+
+      // records remain the same as original
+      getAllRecords should contain theSameElementsAs originalResult
+    }
+
     "add a member successfully" in {
       val groupId = "group-id-1"
       val userIds = Set("user-id-1")
@@ -128,6 +139,18 @@ class MySqlMembershipRepositoryIntegrationSpec
       repo.removeMembers(groupId, toBeRemoved).unsafeRunSync() should contain theSameElementsAs toBeRemoved
       val expectedUserIds = userIds -- toBeRemoved
       getAllRecords should contain theSameElementsAs expectedUserIds.map(Tuple2(_, groupId))
+    }
+
+    "do nothing if the member set is empty" in {
+      val groupId = "group-id-1"
+      val userIds = Set("user-id-1")
+
+      repo.addMembers(groupId, userIds).unsafeRunSync()
+      val originalResult = getAllRecords
+      val result = repo.removeMembers(groupId, Set()).unsafeRunSync()
+      result shouldBe empty
+
+      getAllRecords should contain theSameElementsAs originalResult
     }
   }
 

--- a/modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala
+++ b/modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala
@@ -17,13 +17,12 @@
 package vinyldns.mysql.repository
 
 import org.scalatest._
-import scalikejdbc.DB
+import scalikejdbc.{DB, _}
 import vinyldns.core.domain.membership.MembershipRepository
 import vinyldns.mysql.TestMySqlInstance
-import scalikejdbc._
 
 class MySqlMembershipRepositoryIntegrationSpec
-  extends WordSpec
+    extends WordSpec
     with BeforeAndAfterAll
     with BeforeAndAfterEach
     with Matchers {
@@ -112,7 +111,9 @@ class MySqlMembershipRepositoryIntegrationSpec
       repo.addMembers(groupIdTwo, userIds).unsafeRunSync() shouldBe userIds
 
       val expectedGroups = Set(groupIdOne, groupIdTwo)
-      repo.getGroupsForUser(userIds.head).unsafeRunSync() should contain theSameElementsAs expectedGroups
+      repo
+        .getGroupsForUser(userIds.head)
+        .unsafeRunSync() should contain theSameElementsAs expectedGroups
     }
   }
 
@@ -136,7 +137,9 @@ class MySqlMembershipRepositoryIntegrationSpec
       getAllRecords should contain theSameElementsAs userIds.map(Tuple2(_, groupId))
 
       val toBeRemoved = userIds.take(5)
-      repo.removeMembers(groupId, toBeRemoved).unsafeRunSync() should contain theSameElementsAs toBeRemoved
+      repo
+        .removeMembers(groupId, toBeRemoved)
+        .unsafeRunSync() should contain theSameElementsAs toBeRemoved
       val expectedUserIds = userIds -- toBeRemoved
       getAllRecords should contain theSameElementsAs expectedUserIds.map(Tuple2(_, groupId))
     }
@@ -172,7 +175,9 @@ class MySqlMembershipRepositoryIntegrationSpec
       // not adding to group three
 
       val expectedGroups = Set(groupIdOne, groupIdTwo)
-      repo.getGroupsForUser(underTest.head).unsafeRunSync() should contain theSameElementsAs expectedGroups
+      repo
+        .getGroupsForUser(underTest.head)
+        .unsafeRunSync() should contain theSameElementsAs expectedGroups
     }
 
     "return empty when no groups for user" in {

--- a/modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlMembershipRepository.scala
+++ b/modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlMembershipRepository.scala
@@ -19,9 +19,9 @@ package vinyldns.mysql.repository
 import cats.effect.IO
 import cats.implicits._
 import org.slf4j.LoggerFactory
+import scalikejdbc._
 import vinyldns.core.domain.membership.MembershipRepository
 import vinyldns.core.route.Monitored
-import scalikejdbc._
 
 class MySqlMembershipRepository extends MembershipRepository with Monitored {
   private final val logger = LoggerFactory.getLogger(classOf[MySqlMembershipRepository])
@@ -38,40 +38,46 @@ class MySqlMembershipRepository extends MembershipRepository with Monitored {
     """.stripMargin
 
   def addMembers(groupId: String, memberUserIds: Set[String]): IO[Set[String]] =
-    monitor("repo.Membership.addMembers") {
-      IO {
-        logger.info(s"Saving into group $groupId members $memberUserIds")
-        DB.localTx { implicit s =>
-          val memberUserIdList = memberUserIds.toList
-          val valueClause = " VALUES " + memberUserIdList.as("(?, ?)").mkString(",")
-          val query = BASE_ADD_MEMBERS + valueClause
-          val valueParams: List[String] = memberUserIdList.flatMap(Seq(_, groupId))
-          SQL(query)
-            .bind(valueParams: _*)
-            .update
-            .apply()
+    memberUserIds.toList match {
+      case Nil => IO.pure(memberUserIds)
+      case nonEmpty =>
+        monitor("repo.Membership.addMembers") {
+          IO {
+            logger.info(s"Saving into group $groupId members $nonEmpty")
+            DB.localTx { implicit s =>
+              val valueClause = " VALUES " + nonEmpty.as("(?, ?)").mkString(",")
+              val query = BASE_ADD_MEMBERS + valueClause
+              val valueParams: List[String] = nonEmpty.flatMap(Seq(_, groupId))
+              SQL(query)
+                .bind(valueParams: _*)
+                .update
+                .apply()
 
-          memberUserIds
+              memberUserIds
+            }
+          }
         }
-      }
     }
 
   def removeMembers(groupId: String, memberUserIds: Set[String]): IO[Set[String]] =
-    monitor("repo.Membership.removeMembers") {
-      IO {
-        logger.info(s"Removing from group $groupId members $memberUserIds")
-        DB.localTx { implicit s =>
-          val memberUserIdList = memberUserIds.toList
-          val inClause = " AND user_id IN (" + memberUserIdList.as("?").mkString(",") + ")"
-          val query = BASE_REMOVE_MEMBERS + inClause
-          SQL(query)
-            .bind(groupId :: memberUserIdList: _*)
-            .update()
-            .apply()
+    memberUserIds.toList match {
+      case Nil => IO.pure(memberUserIds)
+      case nonEmpty =>
+        monitor("repo.Membership.removeMembers") {
+          IO {
+            logger.info(s"Removing from group $groupId members $nonEmpty")
+            DB.localTx { implicit s =>
+              val inClause = " AND user_id IN (" + nonEmpty.as("?").mkString(",") + ")"
+              val query = BASE_REMOVE_MEMBERS + inClause
+              SQL(query)
+                .bind(groupId :: nonEmpty: _*)
+                .update()
+                .apply()
 
-          memberUserIds
+              memberUserIds
+            }
+          }
         }
-      }
     }
 
   def getGroupsForUser(userId: String): IO[Set[String]] =

